prov/rxm/src/rxm_init.c

```diff
@@ -106,7 +106,10 @@ int rxm_info_to_core(uint32_t version, const struct fi_info *hints,
 
 		if (hints->domain_attr) {
 			core_info->domain_attr->caps |= hints->domain_attr->caps;
-			core_info->domain_attr->threading = hints->domain_attr->threading;
+			if (rxm_needs_atomic_progress(hints))
+				core_info->domain_attr->threading = FI_THREAD_SAFE;
+			else
+				core_info->domain_attr->threading = hints->domain_attr->threading;
```

Comment:
This is reverting a change made about 2 commits lower in the stack.  Rather than applying a revert commit, I would simply make the changes that are needed.  If this change is needed, then remove it from the prior commit, along with the description from the patch message.

Reply:
Hopefully my response to previous comment makes it clear.